tests/mlmodel_gemini/test_text_generation.py

```diff
@@ -0,0 +1,348 @@
+    set_trace_info()
+    add_custom_attribute("llm.conversation_id", "my-awesome-id")
+    add_custom_attribute("llm.foo", "bar")
+
+    chat = gemini_dev_client.aio.chats.create(model="gemini-2.0-flash")
+    loop.run_until_complete(
+        chat.send_message(
+            message="How many letters are in the word Python?",
+            config=google.genai.types.GenerateContentConfig(max_output_tokens=100, temperature=0.7),
+        )
+    )
+
+
+@reset_core_stats_engine()
+@override_llm_token_callback_settings(llm_token_count_callback)
+@validate_custom_events(add_token_count_to_events(text_generation_recorded_events))
+@validate_custom_event_count(count=3)
+@validate_transaction_metrics(
+    name="test_text_generation:test_gemini_text_generation_async_with_token_count",
+    custom_metrics=[(f"Supportability/Python/ML/Gemini/{google.genai.__version__}", 1)],
+    background_task=True,
+)
+@validate_attributes("agent", ["llm"])
+@background_task()
+def test_gemini_text_generation_async_with_token_count(gemini_dev_client, loop, set_trace_info):
+    set_trace_info()
+    add_custom_attribute("llm.conversation_id", "my-awesome-id")
+    add_custom_attribute("llm.foo", "bar")
+
+    chat = gemini_dev_client.aio.chats.create(model="gemini-2.0-flash")
+    loop.run_until_complete(
+        chat.send_message(
+            message="How many letters are in the word Python?",
+            config=google.genai.types.GenerateContentConfig(max_output_tokens=100, temperature=0.7),
+        )
+    )
+
+
+@reset_core_stats_engine()
+@validate_custom_events(events_sans_llm_metadata(text_generation_recorded_events))
+# One summary event, one system message, one user message, and one response message from the assistant
+@validate_custom_event_count(count=3)
+@validate_transaction_metrics(
+    "test_text_generation:test_gemini_text_generation_async_no_llm_metadata",
+    scoped_metrics=[("Llm/completion/Gemini/generate_content", 1)],
+    rollup_metrics=[("Llm/completion/Gemini/generate_content", 1)],
+    background_task=True,
+)
+@background_task()
+def test_gemini_text_generation_async_no_llm_metadata(gemini_dev_client, loop, set_trace_info):
+    set_trace_info()
+
+    chat = gemini_dev_client.aio.chats.create(model="gemini-2.0-flash")
+    loop.run_until_complete(
+        chat.send_message(
+            message="How many letters are in the word Python?",
+            config=google.genai.types.GenerateContentConfig(max_output_tokens=100, temperature=0.7),
+        )
+    )
+
+
+@reset_core_stats_engine()
+@validate_custom_event_count(count=0)
+def test_gemini_text_generation_async_outside_txn(gemini_dev_client, loop):
+    chat = gemini_dev_client.aio.chats.create(model="gemini-2.0-flash")
+    loop.run_until_complete(
+        chat.send_message(
+            message="How many letters are in the word Python?",
+            config=google.genai.types.GenerateContentConfig(max_output_tokens=100, temperature=0.7),
+        )
+    )
+
+
```

Comment:
Do you have a test in here to exercise the string case (I don't see one but maybe I missed it)?